PE-23: Generate opanapi doc based on XSD to support validation (#3606)
OpenNMS/opennms: docs/modules/development/pages/rest/config_management.adoc

= Config Management

> It is the generic specification of Config Management RESTful API

Reviewer: Could we update this to be a bit more specific on why the user uses this? For example:

Use the Confg Management API to ...

Author: extra comment added

> == GET

Reviewer: ```suggestion
== GETs (Reading Data)
```

Author: done

[options="header", cols="5,10"]
|===
| Resource | Description

> | `/rest/cm/\{configName}` | Returns a list of configIds in JSON / YAML format e.g. ["config1", "config2"]

Reviewer: ```suggestion
| `/rest/cm/\{configName}` | Returns a list of configIds in JSON/YAML format, for example, ["config1", "config2"].
```

Author: done

200: Return: String Array

400: Return: {"MESSAGE": "<ERROR_MESSAGE>"}

> | `/rest/cm/schema/\{configName}` | Returns openapi schema (for swagger UI) in JSON / YAML format.

Reviewer: ```suggestion
| `/rest/cm/schema/\{configName}` | Returns OpenAPI schema (for Swagger UI) in JSON/YAML format.
```

Author: done

200: Return: OpenAPI schema

400: Return: {"MESSAGE": "<ERROR_MESSAGE>"}

> | `/rest/cm/\{configName}/\{configId}` | Returns configuration in JSON / YAML format

Reviewer: ```suggestion
| `/rest/cm/\{configName}/\{configId}` | Returns configuration in JSON/YAML format.
```

Author: done

200: Return: Config Bean 

400: Return: {"MESSAGE": "<ERROR_MESSAGE>"}
|===

> == POST

Reviewer: ```suggestion
== POSTs (Adding Data)
```

Author: done

[options="header", cols="5,10"]
|===
| Resource                    | Description

> | `/rest/cm/\{configName}/\{configId}`             | Add new config (if same config exist, it will return error.)

Reviewer: ```suggestion
| `/rest/cm/\{configName}/\{configId}`             | Add new configuration.
If the same configuration exists, it will return an error. 
```

Author: done

> 200: nothing return

Reviewer: ```suggestion
200: Return nothing
```

Author: done

400: Return: {"MESSAGE": "<ERROR_MESSAGE>"}
|===

> == PUT

Reviewer: ```suggestion
== PUTs (Modifying Data)
```

Author: done

[options="header", cols="5,10"]
|===
| Resource                    | Description

> | `/rest/cm/\{configName}/\{configId}`             | Update config (if config is not exist, it will return error.)

Reviewer: ```suggestion
| `/rest/cm/\{configName}/\{configId}`             | Update configuration.
If configuration does not exist, it will return error.
```

Author: done

> 200: nothing return

Reviewer: ```suggestion
200: Return nothing
```

Author: done

400: Return: {"MESSAGE": "<ERROR_MESSAGE>"}
|===

> == DELETE

Reviewer: ```suggestion
== DELETEs (Removing Data)
```

Author: done

[options="header", cols="5,10"]
|===
| Resource                    | Description

> | `/rest/cm/\{configName}/\{configId}`             | Delete config

Reviewer: ```suggestion
| `/rest/cm/\{configName}/\{configId}`             | Delete configuration.
```

Author: done

> 200: nothing return

Reviewer: ```suggestion
200: Return nothing
```

Author: done

= Config Management

> It is the generic specification of Config Management RESTful API. This set of API is used to GET, ADD, UPDATE, DELETE

Reviewer: ```suggestion
Use the Config Management REST API to manage configurations for OpenNMS.
```

Author: done

> configurations for OpenNMS. Configurations modified by this API will take effect instantly.

Reviewer: Please put second sentence on  a new line.

Author: done
